pkg/apis/core/types.go

```diff
@@ -2605,6 +2612,15 @@ type PodSecurityContext struct {
 	// PodSecurityContext, the value specified in SecurityContext takes precedence.
 	// +optional
 	RunAsNonRoot *bool
+	// Indicates that the container must run as a non-root group.
+	// If true, the Kubelet will validate the image at runtime to ensure that it
+	// does not run as GID 0 (root) and fail to start the container if it does.
+	// If unset or false, no such validation will be performed.
```

Comment:
This should also check the SupplementalGroups and FSGroup.

Reply:
wouldnt that be backward incompatible ?

Reply:
godoc and behavior still needs updating?